packages/anu-vue/src/components/badge/ABadge.tsx

```diff
@@ -0,0 +1,70 @@
+import type { PropType } from 'vue'
+import { computed, defineComponent } from 'vue'
+import { color } from '@/composables/useProps'
+import { isNumeric } from '@/utils/helpers'
+
+export interface AnchorOrigin {
+  vertical: 'top' | 'bottom'
+  horizontal: 'left' | 'right'
+}
+
+export const ABadge = defineComponent({
+  name: 'ABadge',
+  props: {
+    color: {
+      ...color,
+      default: 'primary',
+    },
+    variant: {
+      type: String as PropType<'standard' | 'dot'>,
+      default: 'standard',
+    },
+    max: {
+      type: Number,
+      default: 99,
+    },
+    badgeContent: {
+      type: Number,
+    },
+    anchorOrigin: {
+      type: Object as PropType<AnchorOrigin>,
+      default: () => ({ vertical: 'top', horizontal: 'right' }),
+    },
+    overlap: {
+      type: String as PropType<'rectangular' | 'circular'>,
+      default: 'rectangular',
+    },
+  },
+  setup(props, { slots }) {
+    const formatMaxBadgeContent = (badgeContent: unknown) => {
+      if (!isNumeric(badgeContent))
+        return badgeContent
+
+      const numericBadgeContent = Number(badgeContent)
+      if (numericBadgeContent > props.max)
+        return `${props.max}+`
+
+      return numericBadgeContent
+    }
+
+    const badgeSlotContent = computed(() => {
+      if (props.variant === 'dot')
+        return ''
+      if (slots.badgeContent)
+        return formatMaxBadgeContent(slots.badgeContent?.())
+      if (props.badgeContent)
+        return formatMaxBadgeContent(props.badgeContent)
+
+      return ''
+    })
+
+    return () => <div class={['a-badge-wrapper']}>
```

Comment:
Do you have any idea why `a-badge-wrapper` have extra space when we compare it to it's child?

e.g. In the default demo, the icon have 24x24 and `a-badge-wrapper` have 24x30 🤔 

Adding `display: flex` does solve the issue but it might affect the content